Wrappers/Python/cil/framework/framework.py

```diff
@@ -316,7 +316,7 @@ def allocate(self, value=0, **kwargs):
                 seed = kwargs.get('seed', None)
                 if seed is not None:
                     numpy.random.seed(seed)
-                if dtype in [ numpy.complex , numpy.complex64 , numpy.complex128 ] :
+                if numpy.iscomplexobj(out.array):
```

Comment:
I'm not fond of using `out.array`, but with NEP18 this will probably be ok with different array types? Should we decide to use different ones.

Reply:
I thought we often used `.array` within our DC methods?

If we use different DC backends we might have to define our own set of dtypes anyway. `np.complex64` is probably not universal.

`npcomplexobj()` just needs an array-like object. if the datatypye of any element is complex it returns true.